casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -57,18 +60,23 @@ object BlockAPI {
       DeployServiceResponse(success = false, "Error: Casper instance not available")
     )
 
-  def createBlock[F[_]: Monad: MultiParentCasperRef: Log]: F[DeployServiceResponse] =
+  def createBlock[F[_]: Sync: Monad: MultiParentCasperRef: Log]: F[DeployServiceResponse] =
     MultiParentCasperRef.withCasper[F, DeployServiceResponse](
       casper =>
-        for {
-          maybeBlock <- casper.createBlock
-          result <- maybeBlock match {
-                     case err: NoBlock =>
-                       DeployServiceResponse(success = false, s"Error while creating block: $err")
-                         .pure[F]
-                     case Created(block) => casper.addBlock(block).map(addResponse(_, block))
-                   }
-        } yield result,
+        Monad[F].ifM(Sync[F].delay { createBlockLock.tryLock() })(
```

Comment:
Maybe add a TODO item about using `Bracket`. It should be totally trivial to do because `Sync[F]` extends `Bracket[Throwable, F]`, so you already have all the right capabilities. But the issue right now is because of our conflicting `monix`/`cats`/`cats-effect` dependencies the `Task` instance for `Bracket` doesn't work correctly. 

Regardless though, we don't want to get into the situation we were in before where the lock was not being released under some circumstances.